docs/src/guides/advanced/18_allowlist.md

```diff
@@ -0,0 +1,114 @@
+# Contract Deployment Allowlist
+
+This module restricts contract deployments on zkSync Era to a predefined list of approved addresses.
+
+It consists of:
+
+- `SharedAllowList`: in-memory, async-safe address set.
+- `AllowListTask`: periodically fetches the list from a remote URL.
+- `WhitelistedDeployPoolSink`: enforces the list during L2 transaction submission.
```

Comment:
Bikeshedding: IMO, this is unnecessary implementation detail; node operators could be more interested in how to configure this feature rather than implementation details. Ditto for the "How Allowlist Enforcement Works" section below.